### PR TITLE
Initial SwiftFormat tests

### PR DESCRIPTION
**Pull request by eviltofu**
Initial SwiftFormat testing.

1. [sortedImports], removed [trailing spaces] (empty lines) image 
2. [trailing comma] image 
3. [indent] image

### CODE REVIEW

File: .swiftformat [hunk 1]
```diff
@@ -1,5 +1,20 @@
 #Disable all and explicitly enable them to avoid unintentionally activating new rules in newer versions of SwiftFormat
---disable all
+#--disable all
+
+--rules \
```
Comment (hboon) [inline .swiftformat:4]: @eviltofu shucks, so many files and include rule changes. Can you help select just a few files to illustrate what's changed and force push, so I don't have go through 136 files?

File: AlphaWallet.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -4813,7 +4813,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if [ \"${CONFIGURATION}\" = \"Debug\" ]; then\n    \"${PODS_ROOT}/SwiftFormat/CommandLineTool/swiftformat\" \"$SRCROOT\"\nfi\n";
+			shellScript = "if [ \"${CONFIGURATION}\" = \"Debug\" ]; then\n    \"${PODS_ROOT}/SwiftFormat/CommandLineTool/swiftformat\" \"$SRCROOT/AlphaWalletTests\"\nfi\n";
```
Comment (hboon) [inline AlphaWallet.xcodeproj/project.pbxproj:4816]: And can you change ths to the script approach like we just discussed in the issue?

Comment (eviltofu) [reply AlphaWallet.xcodeproj/project.pbxproj:4816]: You want me to list each test file or can I just include the whole directory?

Comment (hboon) [reply AlphaWallet.xcodeproj/project.pbxproj:4816]: Each file; for now at least, I think we'll only switch to directories when we cover all the files in a directory. In this case since we are going to just select a few for the PR, definitely list the files.

File: .swiftformat (hunk 1 above)
Comment (hboon) [reply .swiftformat:4]: To be clear, I mean like, only 5 `.swift` files instead of ~136

File: AlphaWallet.xcodeproj/project.pbxproj (hunk 2 above)
Comment (eviltofu) [reply AlphaWallet.xcodeproj/project.pbxproj]: I've used the Xcode file lists facility. All the files that are to be SwiftFormatted are in the file `SwiftFormatFiles.xcfilelist`

File: .swiftformat [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
+	indent, \
+	duplicateImports, \
+	sortedImports, \
+	spaceInsideBraces, \
+	spaceInsideBrackets, \
+	spaceInsideComments, \
+	spaceInsideGenerics, \
+	spaceInsideParens, \
+	trailingCommas, \
```
Comment (hboon) [inline .swiftformat:13]: Would you remove `trailingCommas`? Seems to generate an unnecessary diff when we reformat.

**Review by hboon — CHANGES_REQUESTED** [review]
Thanks. Added checkboxes below so it's clearer what are the TODOs:

I tried a few random files to see what it does with indentation with just this to reduce the diff:

> --rules \
> 	indent, \
> 	typeSugar, \
> 	
> --indent 4

- [x] ^ Let's just introduce these first.

and this stood out:

Changed this (`ENS.swift`):

```
}.tryMap { ensName, resolvedAddress -> String in
    //snipped
}.mapError { e in SmartContractError.embeded(e) }
.eraseToAnyPublisher()
```

```
}.tryMap { ensName, resolvedAddress -> String in
    //snipped
}.mapError { e in SmartContractError.embeded(e) }
    .eraseToAnyPublisher()
```

- [ ] The original seems better as the indentation reflects the call targets better. Any way to make it so?

This is a little like the indentations here having meaning, tying the `doSomething()` with the `if`

```
if something {
    doSomething()
}

if something
    doSomething
```

vs. this which suggests (wrongly) that `doSomething` is not part of the `if`:

```
if something
doSomething
```

---

An example that is OK (`RPCServer.swift`):

```
case .custom(let custom):
    return custom.explorerEndpoint
            .flatMap { URL(string: $0) }
            .flatMap { $0.appendingPathComponent("api").absoluteString }
```
1G
to:

```
case .custom(let custom):
    return custom.explorerEndpoint
        .flatMap { URL(string: $0) }
        .flatMap { $0.appendingPathComponent("api").absoluteString }
```

The indentation amount doesn't matter (let's not get to 1-space indentation!) as much as whether it is indented or not, so this is fine.

Could you help to include these 2 files too and amend the PR title/commit message since they are not files in the test suite? I'm sorry if one of the other files in the test suite would already exhibit this behavior for indentation, feel free to use those instead if so.

**Comment by eviltofu** [comment]
> Thanks. Added checkboxes below so it's clearer what are the TODOs:
> 
> I tried a few random files to see what it does with indentation with just this to reduce the diff:
> 
> > --rules 
> > indent, 
> > typeSugar, \
> > --indent 4
> 
> * [ ]  ^ Let's just introduce these first.

What does this mean? I remove the rest of the rules?

**Comment by hboon** [comment]
> What does this mean? I remove the rest of the rules?

Yes. Indentation seems the most disruptive, lets focus on that one first.

**Comment by eviltofu** [comment]
> Thanks. Added checkboxes below so it's clearer what are the TODOs:
> 
> I tried a few random files to see what it does with indentation with just this to reduce the diff:
> 
> > --rules 
> > indent, 
> > typeSugar, \
> > --indent 4
> 
> * [ ]  ^ Let's just introduce these first.
> 
> and this stood out:
> 
> Changed this (`ENS.swift`):
> 
> ```
> }.tryMap { ensName, resolvedAddress -> String in
>     //snipped
> }.mapError { e in SmartContractError.embeded(e) }
> .eraseToAnyPublisher()
> ```
> 
>  ```
> }.tryMap { ensName, resolvedAddress -> String in
>     //snipped
> }.mapError { e in SmartContractError.embeded(e) }
>     .eraseToAnyPublisher()
> ```
If you place the closing brace on the next line, it formats correctly.
```swift
}.mapError { e in SmartContractError.embeded(e)
}.eraseToAnyPublisher()
```

File: .swiftformat [hunk 4]
```diff
@@ -1,5 +1,20 @@
 #Disable all and explicitly enable them to avoid unintentionally activating new rules in newer versions of SwiftFormat
---disable all
+#--disable all
```
Comment (hboon) [inline .swiftformat:2]: We want this to stay uncommented, reason described in the comment

**Comment by hboon** [comment]
> If you place the closing brace on the next line, it formats correctly.

Ah, this looks odd:

```
}.mapError { e in SmartContractError.embeded(e)
}.eraseToAnyPublisher()
```

should write like this instead, which indents with the correct level:

```
}.mapError { e in
    SmartContractError.embeded(e)
}.eraseToAnyPublisher()
```

But yes, looks like the formatting rule works if formatted this way. Good then.

Can you help update `.swiftformat`, removing the other rules that was added (eg. `duplicateImports`), including `--disable all` at the top again and see if we missed anything similar to that indentation issue above? Then we are good to go with adding a few files in the PR.

File: .swiftformat (hunk 4 above)
Comment (eviltofu) [reply .swiftformat:2]: the `--rules` option only allows the [rules](https://github.com/nicklockwood/SwiftFormat#rules) you specify, so this is redundent?
```
And then individually enable just the rules you want. Alternatively, use the--rules argument to only enable the rules you specify:

--rules indent,linebreaks
```

**Comment by eviltofu** [comment]
> ```
> }.mapError { e in
>     SmartContractError.embeded(e)
> }.eraseToAnyPublisher()
> ```
> 
> But yes, looks like the formatting rule works if formatted this way. Good then.
> 

Unfortunately there is no SwiftLint rule to flag this as an error. So we might have to write a regex to find all of these occurrences.

**Comment by eviltofu** [comment]
I'd suggest we next look at
`braces`
`sortedImports`

**Comment by hboon** [comment]
> > ```
> > }.mapError { e in
> >     SmartContractError.embeded(e)
> > }.eraseToAnyPublisher()
> > ```
> > But yes, looks like the formatting rule works if formatted this way. Good then.
> 
> Unfortunately there is no SwiftLint rule to flag this as an error. So we might have to write a regex to find all of these occurrences.

It's ok, we can fix it somehow or as we see it I guess. As long as SwiftFormat doesn't undo it

**Comment by hboon** [comment]
> I'd suggest we next look at
> braces
> sortedImports

Let's get this PR in first and try it with more files without necessarily committing them and we'll see.

**Comment by eviltofu** [comment]
Ok I've pushed the changes to .swiftformat and the files list without the changed swift files. You can then discard any changed swift file after experimenting.

**Comment by hboon** [comment]
> the --rules option only allows the rules you specify, so this is redundent?

Ah, I see. But why replace `--disable all` with `--rules` if they do the same thing? We should always strive to reduce unnecessary changes in a PR. Including unrelated things just makes it slower to merge and a higher chance of introducing problems.

---

Sorry, I wasn't clear, I mean:

* Make the diff for `.swiftformat` just introduce `indent (4)` and `typeSugar`, keeping the commented out `TODO`s in particular. Those are actually things we want to enable, or at least to look again
* Include the 5-7 `.swift` files too
* Flag the PR for review again

And we merge it, then do this outside of the PR:

> try it with more files without necessarily committing them and we'll see.

File: AlphaWallet.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -4728,6 +4728,25 @@
...
+			);
+			inputPaths = (
+			);
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "if [ \"${CONFIGURATION}\" = \"Debug\" ]; then\n    while read F; do\n        \"${PODS_ROOT}/SwiftFormat/CommandLineTool/swiftformat\" $F\n    done < $SCRIPT_INPUT_FILE_LIST_0\nfi\n";
```
[the remark this answers is not in the document]
Comment (eviltofu) [reply AlphaWallet.xcodeproj/project.pbxproj:4749]:
```bash
if [ "${CONFIGURATION}" = "Debug" ]; then
    command_line = ""
    while read F; do
        command_line+="${F} "
    done < $SCRIPT_INPUT_FILE_LIST_0
    echo "${PODS_ROOT}/SwiftFormat/CommandLineTool/swiftformat" $command_line
fi
```

**Review by hboon — APPROVED** [review]
